cross compilation error

Hello, 

Thank you for sharing a great library.
I got error while cross compiling.

```bash
  Compiling deflate64 v0.1.10
   Compiling zip v4.6.1
   Compiling tracing-subscriber v0.3.22
   Compiling which v8.0.0
   Compiling neofetch v0.2.0
error[E0308]: arguments to this function are incorrect
  --> /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/neofetch-0.2.0/src/platform/android.rs:40:9
   |
40 |         __system_property_get(
   |         ^^^^^^^^^^^^^^^^^^^^^
   |
note: expected `*const i8`, found `*const u8`
  --> /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/neofetch-0.2.0/src/platform/android.rs:41:13
   |
41 |             prop_cstr.as_ptr() as *const u8,
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: expected raw pointer `*const i8`
              found raw pointer `*const u8`
note: expected `*mut i8`, found `*mut u8`
  --> /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/neofetch-0.2.0/src/platform/android.rs:42:13
   |
42 |             buffer.as_mut_ptr() as *mut u8,
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: expected raw pointer `*mut i8`
              found raw pointer `*mut u8`
note: function defined here
  --> /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/neofetch-0.2.0/src/platform/android.rs:12:8
   |
12 |     fn __system_property_get(name: *const libc::c_char, value: *mut libc::c_char...
   |        ^^^^^^^^^^^^^^^^^^^^^ ----                       -----

error[E0308]: mismatched types
  --> /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/neofetch-0.2.0/src/platform/android.rs:54:24
   |
54 |         CStr::from_ptr(buffer.as_ptr() as *const u8)
   |         -------------- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `*const i8`, found `*const u8`
   |         |
   |         arguments to this function are incorrect
   |
   = note: expected raw pointer `*const i8`
              found raw pointer `*const u8`
note: associated function defined here
  --> /rustc/56f24e00c62ec2db9a8b407e7aef532caaf71062/library/core/src/ffi/c_str.rs:253:24

error[E0308]: arguments to this function are incorrect
  --> /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/neofetch-0.2.0/src/share.rs:73:9
   |
73 |         __system_property_get(
   |         ^^^^^^^^^^^^^^^^^^^^^
   |
note: expected `*const i8`, found `*mut u8`
  --> /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/neofetch-0.2.0/src/share.rs:74:13
   |
74 |             prop_cstr.as_ptr() as *mut u8,
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: expected raw pointer `*const i8`
              found raw pointer `*mut u8`
note: expected `*mut i8`, found `*mut u8`
  --> /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/neofetch-0.2.0/src/share.rs:75:13
   |
75 |             buffer.as_mut_ptr() as *mut u8,
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: expected raw pointer `*mut i8`
              found raw pointer `*mut u8`
note: function defined here
  --> /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/neofetch-0.2.0/src/share.rs:59:8
   |
59 |     fn __system_property_get(name: *const libc::c_char, value: *mut libc::c_char...
   |        ^^^^^^^^^^^^^^^^^^^^^ ----                       -----

error[E0308]: mismatched types
  --> /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/neofetch-0.2.0/src/share.rs:84:24
   |
84 |         CStr::from_ptr(buffer.as_ptr() as *const u8)
   |         -------------- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `*const i8`, found `*const u8`
   |         |
   |         arguments to this function are incorrect
   |
   = note: expected raw pointer `*const i8`
              found raw pointer `*const u8`
note: associated function defined here
  --> /rustc/56f24e00c62ec2db9a8b407e7aef532caaf71062/library/core/src/ffi/c_str.rs:253:24

For more information about this error, try `rustc --explain E0308`.
error: could not compile `neofetch` (lib) due to 4 previous errors
warning: build failed, waiting for other jobs to finish...
note: If the build failed due to a missing target, you can run this command:
note: 
note:     rustup target install x86_64-linux-android
```

--- a/src/gpu.rs
+++ b/src/gpu.rs
@@ -6,14 +6,14 @@ use human_bytes::human_bytes;
 pub struct Gpu {
     pub name: String,
     pub version: String,
-    pub ram: u32,
+    pub ram: u64,
 }
 impl Display for Gpu {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         let mut v = vec![];
         v.push(self.name.clone());
         if self.ram > 0 {
-            v.push(format!("({})", human_bytes(self.ram)));
+            v.push(format!("({})", human_bytes(self.ram as f64)));
         }
 
         if !self.version.is_empty() {
@@ -22,6 +22,38 @@ impl Display for Gpu {
 
         f.write_str(&v.join(" "))
     }
+}
+
+#[cfg(windows)]
+fn get_gpu_vram_from_registry() -> std::collections::HashMap<String, u64> {
+    use winreg::RegKey;
+    use winreg::enums::*;
+
+    let mut map = std::collections::HashMap::new();
+    let hklm = RegKey::predef(HKEY_LOCAL_MACHINE);
+    let class_key = match hklm.open_subkey(
+        r"SYSTEM\ControlSet001\Control\Class\{4d36e968-e325-11ce-bfc1-08002be10318}",
+    ) {
+        Ok(k) => k,
+        Err(_) => return map,
+    };
+
+    for subkey_name in class_key.enum_keys().filter_map(|k| k.ok()) {
+        if let Ok(subkey) = class_key.open_subkey(&subkey_name) {
+            let mem = match subkey.get_value::<u64, _>("HardwareInformation.qwMemorySize") {
+                Ok(v) => v,
+                Err(_) => continue,
+            };
+            // Index by both DriverDesc and AdapterString for robust matching
+            if let Ok(desc) = subkey.get_value::<String, _>("DriverDesc") {
+                map.insert(desc, mem);
+            }
+            if let Ok(adapter_str) = subkey.get_value::<String, _>("HardwareInformation.AdapterString") {
+                map.insert(adapter_str, mem);
+            }
+        }
+    }
+    map
 }
 
 #[cfg(windows)]
@@ -38,18 +70,25 @@ pub async fn get_gpu() -> Option<Vec<Gpu>> {
         #[serde(rename = "DriverVersion")]
         pub driver_version: String,
         #[serde(rename = "AdapterRAM")]
-        pub adapter_ram: u32,
+        pub adapter_ram: u64,
     }
 
     let results: Vec<VideoController> = wmi_query().await.ok()?;
+    let vram_map = get_gpu_vram_from_registry();
 
     Some(
         results
             .iter()
-            .map(|i| Gpu {
-                name: i.caption.to_owned(),
-                version: i.driver_version.to_owned(),
-                ram: i.adapter_ram,
+            .map(|i| {
+                let ram = vram_map
+                    .get(&i.caption)
+                    .copied()
+                    .unwrap_or(i.adapter_ram);
+                Gpu {
+                    name: i.caption.to_owned(),
+                    version: i.driver_version.to_owned(),
+                    ram,
+                }
             })
             .collect(),
     )

--- a/src/platform/android.rs
+++ b/src/platform/android.rs
@@ -38,8 +38,8 @@ pub fn get_property(property: &str) -> Result<String> {
 
     let result = unsafe {
         __system_property_get(
-            prop_cstr.as_ptr() as *const u8,
-            buffer.as_mut_ptr() as *mut u8,
+            prop_cstr.as_ptr() as *const libc::c_char,
+            buffer.as_mut_ptr() as *mut libc::c_char,
         )
     };
 
@@ -51,7 +51,7 @@ pub fn get_property(property: &str) -> Result<String> {
     }
 
     let value = unsafe {
-        CStr::from_ptr(buffer.as_ptr() as *const u8)
+        CStr::from_ptr(buffer.as_ptr() as *const libc::c_char)
             .to_string_lossy()
             .into_owned()
     };

--- a/src/share.rs
+++ b/src/share.rs
@@ -71,8 +71,8 @@ pub fn get_property(property: &str) -> Option<String> {
 
     let result = unsafe {
         __system_property_get(
-            prop_cstr.as_ptr() as *mut u8,
-            buffer.as_mut_ptr() as *mut u8,
+            prop_cstr.as_ptr() as *mut libc::c_char,
+            buffer.as_mut_ptr() as *mut libc::c_char,
         )
     };
 
@@ -81,7 +81,7 @@ pub fn get_property(property: &str) -> Option<String> {
     }
 
     let value = unsafe {
-        CStr::from_ptr(buffer.as_ptr() as *const u8)
+        CStr::from_ptr(buffer.as_ptr() as *const libc::c_char)
             .to_string_lossy()
             .into_owned()
     };

